### PR TITLE
Named argument detection is scope php-version dependent

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/phpstan/phpstan-src/pull/3642#issuecomment-2495429969

### CODE REVIEW

File: tests/PHPStan/Rules/Functions/data/named-arguments.php [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 namespace FunctionNamedArguments;
-
+if (PHP_VERSION_ID < 80000) return;
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/Functions/data/named-arguments.php]: We don't need this here, right? The test should run only on PHP 8+.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Rules/Functions/data/named-arguments.php (hunk 1 above)
Comment (staabm) [reply tests/PHPStan/Rules/Functions/data/named-arguments.php]: the test which uses this file, has error expectations for PHP 7, see

https://github.com/phpstan/phpstan-src/blob/a0e007aee46a0a3ac56d74789f1107a87d0f5740/tests/PHPStan/Rules/Functions/CallToFunctionParametersRuleTest.php#L494-L499

Comment (ondrejmirtes) [reply tests/PHPStan/Rules/Functions/data/named-arguments.php]: It doesn't make sense to me. PHP 7.x does not support named arguments so it doesn't make sense to check them?

Comment (staabm) [reply tests/PHPStan/Rules/Functions/data/named-arguments.php]: so do you say I should skip the test on php7, or should the test be made to assert 0 errors on php7?
(and I guess the rule needs adjustments for that)

Comment (staabm) [reply tests/PHPStan/Rules/Functions/data/named-arguments.php]: I went with skipping the tests for now

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**